server/src/LDBDWAuth.py

```diff
@@ -224,3 +218,62 @@ def check_authorization_gridmap(self, environ, req_method, full_uri, authorise):
                                 r = admin.log_and_set_http_code(401, c, req_method, "Unable to check authorization in grid-mapfile %s: %s" % (mapfile, e), full_uri)
         # Return.
         return r
+
+class SciTokensAuthorization():
+    def __init__(self):
+        self.admin = Admin.AdminHandle()
+        self.constant = Constants.ConstantsHandle()
+        os.environ['XDG_CACHE_HOME'] = self.constant.scitokens_cache_dir
+        self.token_enforcer = scitokens.Enforcer(self.constant.scitokens_issuer, audience=self.constant.scitokens_audience)
+
+    def check_authorization_scitoken(self, environ, req_method, full_uri, authorise):
+        # Instantiate.
+        # If using HTTP.
+        if not self.constant.use_https or str(environ['LocalAccess'])=='True':
+            # Set result to OK and carry on.
+            r = [200]
+        # Otherwise, using HTTPS.
+        else:
```

Comment:
Can this just `return [200]` then we can drop the `else` and deindent the rest of this (very long) function.

```suggestion
            return [200]
        # Otherwise, using HTTPS.
```